doc/sphinx-guides/source/developers/provenance.rst

```diff
@@ -0,0 +1,85 @@
+==========
+Provenance
+==========
+
+.. contents:: |toctitle|
+    :local:
+
+Introduction
+------------
+
+Support for "provenance" is being added to Dataverse. Provenance is metadata that describes the history of a digital object: where it came from, how it came to be in its present state, who or what acted upon it, etc.
+
+Running the Provenance System as Part of Your Development Environment
+---------------------------------------------------------------------
+
+See the :doc:`dev-environment` section for how to run the prov system as a developer. You will need this system running to work on features related to provenance in Dataverse.
+
+Building the Provenance RPM
+---------------------------
+
+Dataverse maintains RPM spec files to create an RPM of provenance components for installation on RHEL/CentOS.
+
+In the Dataverse source tree, the RPM spec file can be found at ``doc/sphinx-guides/source/_static/developers/prov/dataverse-provenance.spec`` or :download:`downloaded <../_static/developers/prov/dataverse-provenance.spec>`.
+
+We are using Vagrant to build the RPM. Your starting point should be the ``Vagrantfile`` in the root of this git repo, the main repo for Dataverse. In that ``Vagrantfile``, temporarily comment out any shell provisioning lines such as ``config.vm.provision "shell", path: "scripts/vagrant/setup.sh"``. We don't need to install Dataverse to build the prov RPM. We just need a bare-bones CentOS environment onto which we will install the packages we need to build the RPM.
+
+Once you have disabled the setup scripts, start up the VM:
+
+``vagrant up``
+
+Then, ssh into the VM:
+
+``vagrant ssh``
+
+The following steps are all done in the VM. Install the necessary RPMs (add the ``scl`` repo first and separately:
+
+``sudo yum install -y centos-release-scl``
+
+``sudo yum install -y redhat-lsb-core devtoolset-7 boost-devel unixODBC-devel rpm-build rpmdevtools``
+
+We need a newer version of SWIG (3.x) than is available on CentOS 7 (2.x) so we have to install it from source:
+
+``sudo yum install -y pcre-devel python-devel``
+
+``wget wget http://prdownloads.sourceforge.net/swig/swig-3.0.12.tar.gz``
+
+``tar xvfz swig-3.0.12.tar.gz``
+
+``source /opt/rh/devtoolset-7/enable``
+
+``./configure``
+
+``make``
+
+``sudo make install``
+
+Create directories used by rpmbuild:
+
+``mkdir -p ~/rpmbuild/{BUILD,RPMS,SOURCES,SPECS,SRPMS}``
+
+Change to the directory with the spec file:
+
+``cd /dataverse/doc/sphinx-guides/source/_static/developers/prov``
+
+Download the sources:
+
+``spectool -g dataverse-provenance.spec -C ~/rpmbuild/SOURCES``
+
+Build the RPM:
+
+``rpmbuild -ba dataverse-provenance.spec``
```

Comment:
Maybe but I think @landreev and I are the only ones on the team who have ever built an RPM. I purposefully documented the steps in prose. I could see adding more automation once we've gotten more familiar with the process. Unfortunately, we can't use the HMDC Jenkins server to automate the RPM building because it's running el6 and there appear to be incompatibilities between el6 and el7. I left a comment about this at https://github.com/IQSS/dataverse/issues/4364#issuecomment-364464804 . For now the plan is the build this RPM in Vagrant and I don't want to be the only one who knows how. Note also that the version of the RPM matters. At this stage in the game, a human needs to make decisions about when to bump the version and when to get in touch with the upstream author about cutting a release, etc. It's complicated.

Reply:
Ok - I was reminded of the discussions we had in https://github.com/IQSS/dataverse/issues/4369 about documentation / scripts for "surgery" on dependencies, which was why I mentioned it.